document-store/src/main/java/org/hypertrace/core/documentstore/model/options/UpdateOptions.java

```diff
@@ -2,14 +2,21 @@
 
 import static org.hypertrace.core.documentstore.model.options.ReturnDocumentType.AFTER_UPDATE;
 
+import com.mongodb.client.model.FindOneAndUpdateOptions;
 import lombok.Builder;
 import lombok.Value;
 
 @Value
 @Builder
 public class UpdateOptions {
   public static UpdateOptions DEFAULT_UPDATE_OPTIONS =
-      UpdateOptions.builder().returnDocumentType(AFTER_UPDATE).build();
+      UpdateOptions.builder()
+          .returnDocumentType(AFTER_UPDATE)
+          .updateOptions(new com.mongodb.client.model.UpdateOptions())
```

Comment:
let's not strictly couple mongodb UpdateOptions with the genereic document-store update-options

Similar to returnDocumentType can you come up with a specific attribute that only communicates the upsert if not found behaviour